### PR TITLE
Cknieling/update filesystem

### PR DESCRIPTION
**Pull request by chr11115**
removed a few filesystem related util functions in Utilities.cpp/.h with the equivalent functions from std::filesystem

### CODE REVIEW

**Comment by chr11115** [comment]
There are still a few encoding and color related functions left, which were not changed.
I did not trust std::filesystem to handle complex encoding changes, because they state conversion is [unspecified](https://en.cppreference.com/w/cpp/filesystem/path/string)
There is a function for retrieving the modification time of a file in the util functions (getFileModificationTime)
 - this could technically be replaced [last_write_time](https://en.cppreference.com/w/cpp/filesystem/last_write_time) 
 - the return type is [file_time_type](https://en.cppreference.com/w/cpp/filesystem/file_time_type) and not time_t and generally a bit cumbersome to use and print (see [here ](https://eklitzke.org/std-filesystem-last-write-time) and [here](https://stackoverflow.com/questions/56788745/how-to-convert-stdfilesystemfile-time-type-to-a-string-using-gcc-9))
 - if we don't care about debug prints, we could still convert to getFileModificationTime for consistency or keep using stat/_wstat

File: src/serlio/PRTContext.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ PRTContext::PRTContext(const std::vector<std::wstring>& addExtDirs) : mPluginRoo
 	}
 	else {
 		theCache.reset(prt::CacheObject::create(prt::CacheObject::CACHE_TYPE_DEFAULT));
-		mResolveMapCache = std::make_unique<ResolveMapCache>(prtu::getProcessTempDir(SRL_TMP_PREFIX));
+		mResolveMapCache = std::make_unique<ResolveMapCache>(prtu::getProcessTempDir(SRL_TMP_PREFIX).wstring());
```
Comment (mistafunk) [inline src/serlio/PRTContext.cpp:89]: for readability, i would extract `prtu::getProcessTempDir(SRL_TMP_PREFIX)` into a variable here

**Review by mistafunk — CHANGES_REQUESTED** [review]
Found a few things... ;-)

File: src/serlio/utils/Utilities.h [hunk 2]
```diff
@@ -95,19 +96,8 @@ std::vector<const C*> toPtrVec(const std::vector<std::unique_ptr<C, D>>& sv) {
 }
 
 // poor mans std::filesystem - we don't want boost or c++17 dependency right now
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/utils/Utilities.h]: removed ;)

File: src/serlio/utils/ResolveMapCache.cpp [hunk 3]
```diff
@@ -82,7 +82,7 @@ ResolveMapCache::LookupResult ResolveMapCache::get(const std::wstring& rpk) {
 		prt::Status status = prt::STATUS_UNSPECIFIED_ERROR;
 		if (DBG)
 			LOG_DBG << "createResolveMap from " << rpk;
-		rmce.mResolveMap.reset(prt::createResolveMap(rpkURI.c_str(), mRPKUnpackPath.c_str(), &status), PRTDestroyer());
+		rmce.mResolveMap.reset(prt::createResolveMap(rpkURI.c_str(), mRPKUnpackPath.wstring().c_str(), &status), PRTDestroyer());
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/utils/ResolveMapCache.cpp]: added local const wstring

File: src/serlio/utils/ResolveMapCache.cpp [hunk 4]
```diff
@@ -59,10 +59,10 @@ ResolveMapCache::LookupResult ResolveMapCache::get(const std::wstring& rpk) {
 			LOG_DBG << "rpk: cache timestamp: " << it->second.mTimeStamp;
 		if (it->second.mTimeStamp != timeStamp) {
 			mCache.erase(it);
-			std::wstring filename = prtu::filename(rpk);
+			std::filesystem::path filename = std::filesystem::path(rpk).filename();
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/utils/ResolveMapCache.cpp:62]: Noted.

File: src/serlio/PRTContext.cpp (hunk 1 above)
Comment (chr11115) [reply src/serlio/PRTContext.cpp]: Done.

File: src/serlio/materials/MaterialUtils.cpp [hunk 5]
```diff
@@ -186,10 +186,11 @@ std::wstring synchronouslyCreateShadingEngine(const std::wstring& desiredShading
...
-		std::wstring p = shadersPath + L"serlioShaderStingray.sfx";
-		LOG_DBG << "stingray shader located at " << p;
-		return p;
+		const std::wstring shaderPath =
+		        (PRTContext::get().mPluginRootPath.parent_path() / L"shaders/serlioShaderStingray.sfx")
+		                .generic_wstring();
+		LOG_DBG << "stingray shader located at " << shaderPath;
+		return shaderPath;
 	}();
 	return sfxFile;
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/materials/MaterialUtils.cpp:193]: Done.

**Comment by chr11115** [comment]
requested changes were made

File: src/serlio/utils/Utilities.cpp [hunk 6]
```diff
@@ -61,62 +62,17 @@ std::wstring getPluginRoot() {
 		throw std::runtime_error("failed to get plugin location");
 	}
 
-	_splitpath_s(dllPath, drive, 8, dir, _MAX_PATH, 0, 0, 0, 0);
-	std::wstring rootPath = prtu::toUTF16FromOSNarrow(drive);
-	rootPath.append(prtu::toUTF16FromOSNarrow(dir));
+	rootPath = std::filesystem::path(dllPath).parent_path();
```
Comment (mistafunk) [inline src/serlio/utils/Utilities.cpp:65]: please remove the unused variables `drive` and `dir`

**Review by mistafunk — CHANGES_REQUESTED** [review]
overall LGTM, feel free to merge after fixing the unused variables warning ;-)

File: src/serlio/utils/Utilities.cpp (hunk 6 above)
Comment (chr11115) [reply src/serlio/utils/Utilities.cpp:63]: removed unused variables

**Pull request merged**